src/EFCore.PG/Storage/Internal/Mapping/NpgsqlArrayArrayTypeMapping.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Text;
+using Microsoft.EntityFrameworkCore.Storage;
+using System.Diagnostics;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.Mapping
+{
+    /// <summary>
+    /// Maps PostgreSQL arrays to .NET arrays. Only single-dimensional arrays are supported.
+    /// </summary>
+    /// <remarks>
+    /// Note that mapping PostgreSQL arrays to .NET List{T} is also supported via <see cref="NpgsqlArrayListTypeMapping"/>.
+    /// See: https://www.postgresql.org/docs/current/static/arrays.html
+    /// </remarks>
+    public class NpgsqlArrayArrayTypeMapping : NpgsqlArrayTypeMapping
```

Comment:
The problem is that this isn't about collections (which have a specific meaning in .NET)... The idea here was to include both the name of the .NET type and the PG type, since type mappings link the two. I wouldn't be too concerned about naming here - this is pretty internal types that users shouldn't ever encounter...